pagai/services/postgres/main.py

```diff
@@ -122,28 +135,45 @@ def get_table_names(connection=None):
 
     for partition in partitions:
         partition_table_names = partition[0].split(", ")
-        all_table_names = list(
-            filter(lambda x: x[0] not in partition_table_names, all_table_names)
-        )
+        all_table_names = list(filter(lambda x: x[0] not in partition_table_names, all_table_names))
 
     tables = np.array(all_table_names).T[0]
 
     return tables
 
 
 @cache
-def get_table(table, connection=None, limit=1000):
+def explore(table, connection, limit, schema=None):
+    """
+    Returns the first rows of a table alongside the column names.
+    """
+    try:
+        return {
+            "fields": get_column_names(table, schema=schema, connection=connection).tolist(),
+            "rows": get_table(table, schema=schema, connection=connection, limit=limit).tolist(),
+        }
+    except psycopg2.OperationalError as e:
+        raise OperationOutcome(e)
+    except psycopg2.errors.UndefinedTable:
+        raise OperationOutcome(f"Table {table} does not exist in database")
+
+
+@cache
+def get_table(table, connection=None, limit=1000, schema=None):
     """
     Return content of a table with a limit
     """
+    if schema:
+        table = f'"{schema}"."{table}"'
```

Comment:
Maybe we should harmonize our notations... What is called schema here is called owner in fhirpipe (not sure which one is the best, though)

Reply:
I agree, and I believe the right terminology is "schema" because when I do`select * from information_schema.columns`, the columns I get are `table_catalog` `table_schema` `table_name` `column_name`... no mention of "owner" anywhere. I think a schema OWNS a table, therefore you could think of the schema as the "owner", but it's misleading. I used to refer it as "schema" at my previous company.

Reply:
Ok, I'll change the notation in Fhirpipe in a next PR, then